### DISCUSSION TITLE
Various Runs and GLMSingle

### DISCUSSION THREAD

**Discussion by max-kat** [cvnlab/GLMsingle#112] [Q&A]
Hi :) 

I recently came across your paper in eLife and thought GLMSingle is a great idea. Now I have a dataset in which I have to use an RSA and thought I give GLMSingle a try. Since I am a newbie to GLMSingle, I have some basic questions and would like your opinion on certain steps:)

**Study design:** 
Each participant has one session (scan day), consisting of 4 individual runs (continuous scans) 
In each run participants, were presented with 2 types of stimuli (A&B) for 15 seconds and rated them either high or low (in a nutshell) 
Thus, I have 4 conditions A_high, A_low, B_high, B_low.
After each stimulus A&B, there is an additional questionblock that is **not** of interest for this analysis

**Preprocessing based on the original eLife paper:**
1. B0 unwarping
2. Volume realignment in each run (without regressing out motion parameters) 
3. Grand mean scaling (using FSL) 

NOT performed:
smoothing, temporal filtering 

2 Questions:
A. Judging from the code of GLMSingle, I probably need to coregister/align the 4 individual runs per participant to e.g. run 1 of that person, right? Or does GLMSingle take care of this?
B. If I need to align to e.g. run 1, does GLMSingle model 4 individual scanner drifts that occurred during each session or how does it handle the concatenated data? 
C. GLMSingle uses everything that is not coded as baseline, right? So, if I am not interested in several parts of my design (questions after each stimulus), can I just put them all into one single big Condition of no interest, so GLMSingle does not count it as a baseline? Since these time windows of no interest can have differing lengths, does this raise an error if I put them all into one condition of no interest, or is that fine? (Alternatively, can I also just tell GLMSingle to ignore these time points?)

Thank you so much for your time and help! 

Best, 
max

**Comment by kendrickkay** [comment]
A. yes, register the runs to each other. GLMsingle doesn't do anything "spatially" in that sense.

B. Not entirely sure what you mean by scanner drift. GLMsingle allows for signal drift in the sense that the overall intensity in a voxel might drift up or down (for some reason).  It includes a separate set of polynomials for each run in order to attempt to model the (possibly drifting) baseline signal level.

C. Yes, anything not explicitly modeled as an experiment event is treated as the "baseline signal".   In terms of dealing with your question period, you could (i) add a condition and act like it is "repeated" every time a question period occurs, or (ii) You could explicitly include an <extraregressors> input (and construct this regressor yourself) which is then included in the model, as an attempt to account for some of the variance in the time series due to the question period.  As you point out, if neural activity may extend for different lengths of time, then this poses some potential complications.  Whether it matters depends on the numbers and the convolution with the slow HRF...    However, if you go down path (i), then you have to consider the issue of different stimulus durations.  This is discussed on the GLMsingle FAQ  https://glmsingle.readthedocs.io/en/latest/wiki.html  .  For that reason, perhaps it's easier to just dump the variance you believe might be due to the question period into an 'extraregressors' input.

**Reply by max-kat** [reply]
Thank you so much for the quick response! 

ad C, I will construct a regressor for each window of NO interest and let GLMSingle just model it and then ignore the output, cause this should not bias the other regressors of interest, right?

cheers, max

**Comment by kendrickkay** [comment]
Well, "should not bias" is a tricky concept and the meaning of that is tricky to specify...

What we can say is that when you supply regressors in the 'extraregressors' input, what the modeling attempts to do is to assign some variance to the extra regressors and some variance to the regressors you actually care about.  So, yes, it is attempting to "account for" the stuff of no interest.

**Comment by max-kat** [comment]
@kendrickkay thanks so much for your help! I appreciate it! 

I read through this post [https://github.com/cvnlab/GLMsingle/discussions/26] and now wonder how exactly I should best implement the error term modeling <extra_regrossors> if I want to dump all the trials of no interest into a few global regressors:

First:
The 2 conditions A & B are actually of slightly different lengths 13 vs. 15 seconds but since GLMSingle is not made for modelling different stimdur lengths I just use the stimdur of 13 seconds and 'dump' the remaining 2 seconds of each B trial into a global regressor (**remaining_time**), made to soak up this variance. Does this make sense regarding the GLMSingle restrictions of stimdur?

Secondly:
There is no meaningful difference between the 2 question blocks (Q1 & Q2) following conditions A and B but they are slightly different in length 23 vs. 25 seconds. If I want to create ONE <extra_regressor> containing the 2 question blocks (**question_block**), do I need to just indicate each TR where this regressor is present with a one or am I mistaken?  So let's assume a case (TR=1) in which there is the sequence of A (13 sec) Q1(23 sec) B (15 sec) Q2(25 sec), would the regressor design for **question_block** look like 13x0 (A) 23x1(Q1) 13x0 (B) 2x0(**remaining_time**) 25x1(Q2)? and the one for **remaining_time**: 13x0 (A) 23x0(Q1) 13x0 (B) 2x1(**remaining_time**) 25x0(Q2)? 

Thirdly:
In the above example will the nuisance regressor be convolved with an HRF or do I need to time_shift its onset to account for the lag in HRF? 

Fourth:
additionally, some participants have missing trials in A & B (did not provide fast enough feedback). My thinking is since both trials are quite different it might make sense to dump them in different regressors (**A_missing** & **B_missing**).

So all in all my experiment would have 4 global extra regressors that are designed to soak up the events of no interest. 
1. remaining_time: 2 sec of each B trial
2. question_block: the 23 & 25 seconds of each question block 
3. A_missing: all the missing events of A 
4. B_missing: all the missing events of B 

Since you wrote in the aforementioned thread to use 1 -2 extra regressors, I am unsure if my 4 regressor design would introduce some unwanted bias (overfitting) or hinder GLMSingle from making an accurate calculation of the design itself. Is this fear warranted? 

thanks again!:)

**Reply by kendrickkay** [reply]
Regarding the 'remaining_time' issue, the difference between 13s and 15s is small and is likely not going to make any measureable difference. To avoid complexity, I would just choose, say, 14 s as a compromise and use it for both.

For the extra regressors, note that you (the user) construct them fully.  GLMsingle will not try to do any HRF convolution on what you supply -- it will use them as is.

I think you should feel free to use 4 extra regressors if you think they are important to accurately model the data. The 1-2 I mentioned was just a guess since I am not entirely sure about your paradigm.

In general, the most effective approach will probably be to carefully construct your regressors, making sure to visualize them, and ideally against some actual example voxel/ROI time series to make sure you are doing things well.
